edfio/edf.py

```diff
@@ -745,7 +745,9 @@ def _set_num_data_records_with_signals(
         if not signals:
             num_data_records = 1
         else:
-            signal_durations = [len(s._digital) / s.sampling_frequency for s in signals]
+            signal_durations = [
+                round(len(s._digital) / s.sampling_frequency, 12) for s in signals
+            ]
             if any(v != signal_durations[0] for v in signal_durations[1:]):
```

Comment:
```suggestion
            if any(abs(v - signal_durations[0]) > tolerance for v in signal_durations[1:]):
```
Introduce a small tolerance (maybe one percent or even less than the smallest sample duration).